pkg/pgmodel/model/interface.go

```diff
@@ -47,6 +47,9 @@ func MetricTableName(conn pgxconn.PgxConn, metric string) (string, bool, error)
 	var possiblyNew bool
 	defer res.Close()
 	if !res.Next() {
+		if res.Err() != nil {
```

Comment:
why was this hit? is the metric handling goroutine shutting down and then starting up again?

Reply:
Its hit because the connection is blocked and the way pgx deals is lazily returning the block error only after you try reading the row. 

That's why we need to check the error if there are no rows.